src/components/select/multi-select/multi-select.test.js

```diff
@@ -0,0 +1,1318 @@
+        .parent()
+        .should("have.attr", "data-element", testPropValue);
+    });
+
+    it("should render Multi Select with data-role prop set to test value", () => {
+      CypressMountWithProviders(
+        <MultiSelectComponent data-role={testPropValue} />
+      );
+
+      selectElementInput()
+        .parent()
+        .parent()
+        .should("have.attr", "data-role", testPropValue);
+    });
+
+    it.each([["top"], ["bottom"], ["left"], ["right"]])(
+      "should render the help tooltip in the %s position",
+      (tooltipPositionValue) => {
+        CypressMountWithProviders(
+          <MultiSelectComponent
+            labelHelp="Help"
+            tooltipPosition={tooltipPositionValue}
+          />
+        );
+
+        helpIcon().trigger("mouseover");
+        tooltipPreview()
+          .should("be.visible")
+          .and("have.css", tooltipPositionValue);
+      }
+    );
+
+    it("should check Multi Select is disabled", () => {
+      CypressMountWithProviders(<MultiSelectComponent disabled />);
+
+      commonDataElementInputPreview()
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should render Multi Select as read only", () => {
+      CypressMountWithProviders(<MultiSelectComponent readOnly />);
+
+      commonDataElementInputPreview().should("have.attr", "readOnly");
+      selectInput().click();
+      selectList().should("not.exist");
+    });
+
+    it.each([
+      ["small", "32px"],
+      ["medium", "40px"],
+      ["large", "48px"],
+    ])(
+      "should use %s as size and render Multi Select with %s as height",
+      (size, height) => {
+        CypressMountWithProviders(<MultiSelectComponent size={size} />);
+
+        commonDataElementInputPreview()
+          .parent()
+          .should("have.css", "min-height", height);
+      }
+    );
+
+    it("should check Multi Select has autofocus", () => {
+      CypressMountWithProviders(<MultiSelectComponent autoFocus />);
+
+      commonDataElementInputPreview().should("be.focused");
+    });
+
+    it("should check Multi Select is required", () => {
+      CypressMountWithProviders(<MultiSelectComponent required />);
+
+      verifyRequiredAsteriskForLabel();
+    });
+
+    it("should check Multi Select label is inline", () => {
+      CypressMountWithProviders(<MultiSelectComponent labelInline />);
+
+      getDataElementByValue("label")
+        .parent()
+        .should("have.css", "-webkit-box-pack", "end");
+    });
+
+    it.each([
+      ["inline", "399"],
+      ["inline", "400"],
+      ["above", "401"],
+    ])(
+      "should check Multi Select label is %s with adaptiveLabelBreakpoint %s and viewport 400",
+      (alignment, breakpoint) => {
+        cy.viewport(400, 300);
+
+        CypressMountWithProviders(
+          <MultiSelectComponent
+            labelInline
+            adaptiveLabelBreakpoint={breakpoint}
+          />
+        );
+
+        if (alignment === "inline") {
```

Comment:
It's more like a question than a review message, wondering is there any reason that we need to get rid of the if statement here? I think Sian has this in her filterable-select test and it makes sense of having the if here, so why do we try to use another parameter to avoid if?

Reply:
@ZhuoyuJin on reflection, it most likely uses more resources to cycle around the `if` statement in this case than it would to check one more variable in the array so I've amended both Multi and Filterable Select with @DlgSHi 's suggested change.